alfred-telemetry-solr/solr6/src/main/java/eu/xenit/alfred/telemetry/solr/handler/PrometheusSummaryHandler.java

```diff
@@ -0,0 +1,298 @@
+                objectName = new ObjectName((String) beanToMonitor.getKey());
+            } catch (MalformedObjectNameException e) {
+                logger.error("Malformed bean name " + beanToMonitor.getKey());
+                continue;
+            }
+            MBeanInfo mBeanInfo = null;
+            try {
+                mBeanInfo = mbeanServer.getMBeanInfo(objectName);
+            } catch (InstanceNotFoundException | IntrospectionException | ReflectionException e) {
+                logger.error("Problem with bean " + beanToMonitor.getKey());
+                continue;
+            }
+            for (MBeanAttributeInfo attributeInfo : mBeanInfo.getAttributes()) {
+                ArrayList values = (ArrayList) beanToMonitor.getValue();
+                if (prometheusAllowedType(attributeInfo.getType()) && ("*".equals(values.get(0)))) {
+                    String resp = String.format("%s_%s{%s}", getPrometheusName(objectName), attributeInfo.getName(),
+                            getPrometheusLabels(objectName));
+                    try {
+                        rsp.add(resp, mbeanServer.getAttribute(objectName, attributeInfo.getName()));
+                    } catch (MBeanException | AttributeNotFoundException | InstanceNotFoundException | ReflectionException e) {
+                        logger.error("Problem with attributes " + attributeInfo + " checked against " + values);
+                        continue;
+                    }
+                }
+            }
+        }
+    }
+
+    private String getPrometheusLabels(ObjectName objectName) {
+        String result = "";
+        for(String property : objectName.getKeyPropertyList().keySet()) {
+            if(!"type".equals(property)) {
+                result += getPrometheusEscape(property) + "=" + getPrometheusQuoted(objectName.getKeyPropertyList().get(property)) + ",";
+            }
+        }
+        if(result.length()>0)
+            return result.substring(0,result.length()-1);
+        else
+            return result;
+    }
+
+    private String getPrometheusQuoted(String s) {
+        return (!s.startsWith("\"")) ? "\"" + s + "\"" : s;
+    }
+
+    private String getPrometheusEscape(String property) {
+        return property.replace(" ",PROMETHEUS_SEPARATOR).replace(".",PROMETHEUS_SEPARATOR).replace("/",PROMETHEUS_SEPARATOR);
+    }
+
+    private String getPrometheusName(ObjectName objectName) {
+        String result = getPrometheusEscape(objectName.getDomain());
+        result += PROMETHEUS_SEPARATOR + getPrometheusEscape(objectName.getKeyPropertyList().get("type"));
+        return result;
+    }
+
+    private boolean prometheusAllowedType(String type) {
+        return ("int".equals(type) || "double".equals(type) || "long".equals(type) ||
+                "java.lang.Integer".equals(type) || "java.lang.Double".equals(type) || "java.lang.Long".equals(type));
+    }
+
+    private void getTrackerMetrics(SolrQueryRequest req, SolrQueryResponse rsp) {
```

Comment:
what do the metric names look like ?
the JVM metrics etc should be aligned with those from `alfred-telemetry-platform`

Reply:
I think that will be difficult, since they don't come from the same source. A mapping is possible only if metrics are the same, I will check how far we are from this.

Reply:
Might be easier to integrate micrometer for jvm metrics.

Reply:
Ok, lets get this merged first and then look at that as a next step.